vllm/v1/worker/gpu_worker.py

```diff
@@ -210,6 +231,290 @@ def profile(self, is_start=True):
+        self.worker.load_model()
+
+    @staticmethod
+    def make_worker_process(
+            vllm_config: VllmConfig,
+            local_rank: int,
+            rank: int,
+            distributed_init_method: str,
+            input_shm_handle,  # Receive SchedulerOutput
+    ) -> WorkerProcHandle:
+        context = get_mp_context()
+
+        # ZMQ paths for initialization-related messages between this process and
+        # the worker process being created. After process creation, the worker
+        # sends a ready message to the output_path. Subsequent messages are sent
+        # back and forth in a call-and-response manner.
+        initialization_input_path = get_open_zmq_ipc_path()
+        initialization_output_path = get_open_zmq_ipc_path()
+
+        process_kwargs = {
+            "vllm_config": vllm_config,
+            "local_rank": local_rank,
+            "rank": rank,
+            "distributed_init_method": distributed_init_method,
+            "input_shm_handle": input_shm_handle,
+            "initialization_input_path": initialization_output_path,
+            "initialization_output_path": initialization_input_path,
+        }
+        # Run EngineCore busy loop in background process.
+        proc = context.Process(target=WorkerProc.run_worker,
+                               kwargs=process_kwargs,
+                               daemon=True)
+        proc.start()
+
+        # Wait for startup
+        model_output_mq_handle = WorkerProc.wait_for_startup(
+            proc, initialization_input_path)
+
+        return WorkerProcHandle(proc, initialization_input_path,
+                                initialization_output_path,
+                                model_output_mq_handle)
+
+    def shutdown(self):
+        self.worker_request_mq = None
+        self.model_output_mq = None
+        destroy_model_parallel()
+        destroy_distributed_environment()
+
+    @staticmethod
+    def run_worker(*args, **kwargs):
+        """ Worker initialization and execution loops.
+        This runs a background process """
+
+        # Signal handler used for graceful termination.
+        # SystemExit exception is only raised once to allow this and worker
+        # processes to terminate without error
+        shutdown_requested = False
+
+        def signal_handler(signum, frame):
+            nonlocal shutdown_requested
+            if not shutdown_requested:
+                shutdown_requested = True
+                raise SystemExit()
+
+        # Either SIGTERM or SIGINT will terminate the worker
+        signal.signal(signal.SIGTERM, signal_handler)
+        signal.signal(signal.SIGINT, signal_handler)
+
+        worker = None
+        try:
+            worker = WorkerProc(*args, **kwargs)
+            worker.model_initialization_loop(
+                kwargs["initialization_input_path"],
+                kwargs["initialization_output_path"])
+
+            worker.execute_model_busy_loop()
```

Comment:
having a busy loop is fine, but please do not only use it for model execution, it will not be extensible, e.g.:

- loading / unloading lora
- RLHF usecase I'm working on, where we need to perform some weight update and then continue execution.

Reply:
Yes, I will add other message types. I think this should be done in a subsequent PR